tests/nn/data_parallel/test_sharded_ddp.py

```diff
@@ -123,6 +115,84 @@ def run_test(backend, device, world_size=2):
     mp.spawn(run_one_step, args=(world_size, backend, device, temp_file_name), nprocs=world_size, join=True)
 
 
+def test_step_on_cpu():
+    run_test(backend=dist.Backend.GLOO, device=torch.device("cpu"), world_size=4)
+
+
+@skip_if_no_cuda
+@skip_if_single_gpu
+def test_step_on_gpu():
+    run_test(backend=dist.Backend.NCCL, device=torch.device("cuda"))
+
+
+def run_ddp_parity(rank, world_size, backend, temp_file_name):
```

Comment:
new unit test which checks that we have the same results as DDP, so that this does not happen again